arangodb-net-standard/Transport/Http/HttpApiTransport.cs

```diff
@@ -132,6 +134,27 @@ public async Task<IApiClientResponse> PatchAsync(string requestUri, StringConten
             return new HttpApiClientResponse(response);
         }
 
+        /// <summary>
+        /// Send a HEAD request using <see cref="HttpClient"/>
+        /// </summary>
+        /// <param name="requestUri"></param>
+        /// <param name="headers"></param>
+        /// <returns></returns>
+        public async Task<IApiClientResponse> HeadAsync(string requestUri, WebHeaderCollection webHeaderCollection = null)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Head, requestUri);
+
+            for (int i = 0; i < webHeaderCollection.Count; i++)
```

Comment:
You risk `NullReferenceException` here if `webHeaderCollection` is null, which is the default value.

You may also consider enumerating using

 ```
foreach (var key in webHeaderCollection.AllKeys) 
{
  request.Headers.Add(key, webHeaderCollection[key]);
}
```

Reply:
But you really do need to check for `null` I think